packages/xchain-ethereum/__tests__/client.testnet.test.ts

```diff
@@ -383,15 +383,15 @@ describe('Client Test', () => {
     )
     mock_thornode_inbound_addresses_success(thornodeApiUrl, [
       {
-        chain: 'ETH',
+        chain: 'ETH' as Chain,
```

Comment:
Before introducing `enum Chain` we were safe to use 'ETH' or any other "string" defined in`xchain-util/src/chain.const.ts`. Now we have to cast it using `as` (which is unsafe). From my perspective we don't need `enum Chain`.

Reply:
A TypeScript `enum` is just syntactic sugar for an exported type and an exported object with the same name:
```
export enum Foo {
  Bar = 'bar'
  Baz = 'baz'
}
```
is equivalent to
```
export type Foo = 'bar' | 'baz'
export const Foo = {
  Bar: 'bar',
  Baz: 'baz',
}
```

Enums thus provide two distinct advantages over plain union types:

First, they provide not only the union type but a mapping object that exists at runtime Note, for example, how this is used in the `singleFee()` helper:
```
export function singleFee(feeType: FeeType, amount: Fee): Fees {
  return Object.values(FeeOption).reduce<Partial<Fees>>((a, x) => ((a[x] = amount), a), {
    type: feeType,
  }) as Fees
}
```
Here, `Object.values(FeeOption)` won't work if FeeOption is only the name of a type.

Second, the exported object effectively provides a "namespace" for the enum values. For example:
```
enum Foo {
    Foo = 'foo',
    Bar = 'bar',
}

enum Bar {
    Foo = 'foo',
    Bar = 'bar',
}

let foo: Foo = Foo.Foo
foo = Foo.Bar
foo = Bar.Foo // Type 'Bar.Foo' is not assignable to type 'Foo'.ts(2322)
```

This is a good thing, because it makes you be more careful about where you're getting your values from. There are a lot of external libraries using the strings `mainnet` and `testnet` as inputs and outputs, for example, and they may not mean exactly the same thing in each context. Using an enum forces you to either use `Network.Mainnet` or `'mainnet' as Network` -- and either way forces you to be upfront that yes, you know what you're doing, and that you _do_ mean _that_ specific kind of `'mainnet'`.


Note that the casts are only used in `*.test.ts`; everywhere else direct references to the enum members (i.e `Chain.Ethereum` or `Network.Mainnet` or such) have been used; I did this specifically so that the tests would provide assurance that the change does not break the existing API for clients that already use literal strings.

Reply:
Thanks for this detailed explanation ^. Let's go with `enum` then.